background_scripts/JobQueue.js

```diff
@@ -0,0 +1,118 @@
+class JobDemon {
+  /**
+   * @param {string} url Url d'un job
+   */
+  constructor(url) {
+    /** @type {string} */
+    this.url = url;
+    /** @type {HTMLIFrameElement} */
+    this.iframe = document.createElement("iframe");
+    /** @type {number} */
+    this.lastExecutionTimestamp = Date.now();
+
+    // On charge l'URL dans une iframe
+    // Ici on laisse la main au content script qui va vérifier si un RDV est disponible
+    this.iframe.src = url;
+
+    document.body.appendChild(this.iframe);
+  }
+
+  retry() {
+    this.iframe.contentWindow.postMessage({ type: "retry" }, "*");
+    this.lastExecutionTimestamp = Date.now();
+  }
+
+  kill() {
+    this.iframe.remove();
+    this.iframe = null;
+  }
+}
+
+class JobQueue {
+  /**
+   * @param {number} delayBetweenJobs Delais entre deux jobs, en secondes
+   * @param {number} delayRetryJob Delais entre deux executions d'un même jobs, en secondes
+   * @param {(job: string) => void} onJobStart Callback quand un job débute
+   */
+  constructor(delayBetweenJobs, minDelayRetryJob, onJobStart) {
+    /** @type {number} en millisecondes */
+    this.delayBetweenJobs = delayBetweenJobs * 1000;
+    /** @type {number} en millisecondes */
+    this.delayRetryJob = minDelayRetryJob * 1000;
+    /** @type {(job: string) => void} Callback quand un job débute */
+    this.onJobStart = onJobStart;
+    /** @type {number} */
+    this.intervalRef = null;
+    /** @type {Object<string, JobDemon>} map de deamon traitant les jobs */
+    this.demons = {};
+    /** @type {string[]} tableau d'url de job */
+    this.jobs = [];
```

Comment:
Pas besoin de changer pour cette PR, mais pour info la plupart des navigateurs supportent une syntaxe un peu moins lourde pour l'initialisation de variables d'instance: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Public_class_fields
(évidemment ça ne s'applique pas aux variables qui dépendent des paramètres du constructeur)

Reply:
Ha ben j'ai essayé, j'me suis fait jeté par un des linter de mémoire :(

Reply:
Je confirme, ca plait pas à web-ext lint... J'avais tout fait comme ca, j'ai du tout changer...

`JS_SYNTAX_ERROR   JavaScript syntax error (Parsing as         There is a JavaScript syntax error in your code, which might be related to some    background_scripts/JobQueue.js   32     8     
                  module error: Unexpected token = at line:   experimental JavaScript features that aren't an official part of the language
                  32 and column: 8) (Parsing as script        specification and therefore not supported yet. The validation cannot continue on
                  error: Unexpected token = at line: 32 and   this file.
                  column: 8)`